src/libmongoc/tests/test-mongoc-sdam.c

```diff
@@ -215,6 +216,250 @@ test_sdam_cb (bson_t *test)
    mongoc_client_destroy (client);
 }
 
+/* Initialize a test context to run one SDAM integration test file.
+ *
+ * Do not use json_test_ctx_init to initialize a context. It sends commands to
+ * check for sessions support. That interferes with failpoints set on isMaster.
+ * Plus, SDAM tests have other special requirements that make initializing the
+ * test context a two step process (explained in run_one_integration_test).
+ */
+static void
+sdam_json_test_ctx_init_config (json_test_ctx_t *ctx,
+                                const json_test_config_t *config)
+{
+   memset (ctx, 0, sizeof (*ctx));
+   ctx->config = config;
+   bson_init (&ctx->events);
+   ctx->acknowledged = true;
+   ctx->verbose = test_framework_getenv_bool ("MONGOC_TEST_MONITORING_VERBOSE");
+   bson_init (&ctx->lsids[0]);
+   bson_init (&ctx->lsids[1]);
+   bson_mutex_init (&ctx->mutex);
+}
+
+static void
+sdam_json_test_ctx_init_client (json_test_ctx_t *ctx, mongoc_client_t *client)
```

Comment:
Is there any reason to split this up from `sdam_json_test_ctx_init_config` vs. having a single `sdam_json_test_ctx_init` function? They always seem to be called together.

Perhaps this is necessary because of the comment you have below in `run_one_integration_test`: "Initializing the test context in tandem with monitoring is a little tricky."